### PR TITLE
release 0.4.1 build recipe update

### PR DESCRIPTION
**Pull request by orbeckst**
~Bug fix release~

EDIT: This PR tries to implement a suggestion by the linter, namely https://conda-forge.org/docs/maintainer/adding_pkgs.html#use-pip

### CODE REVIEW

**Comment by orbeckst** [comment]
@kain88-de was PR  #5 automatically generated? When did that happen? I followed the instructions in https://github.com/MDAnalysis/GridDataFormats/blob/master/MAINTAINER.md which said to create a PR. Do we have to change the instructions?

**Comment by richardjgowers** [comment]
Conda gorge sees tag pushes and makes the PR, so yes instructions are outdated   
  

  
  

  
  
>

**Comment by orbeckst** [comment]
PR #5 omitted https://github.com/conda-forge/griddataformats-feedstock/pull/6/commits/548b1c07cb7241b27eab0a0ed502ac980b98af71 (to make the linter happier) so how can I do this now? Manually create a PR and increase the build number?

**Comment by richardjgowers** [comment]
@conda-forge-admin, please rerender

**Comment by richardjgowers** [comment]
@orbeckst yeah incrementing build number will make a new build of the same version.  I tried off this PR but the diff looks noisy, might need a rebase or something.

**Comment by kain88-de** [comment]
Conda forge has a new update bot since last year. It automatically creates
a pr when there was an update on pypi.

**Comment by orbeckst** [comment]
@richardjgowers I rebased against master and increased the build number

**Comment by orbeckst** [comment]
I have no idea why the _run docker build_ stage failed https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=26937 
```
Attempting to finalize metadata for griddataformats
Error: Failed to render jinja template in /home/conda/recipe_root/meta.yaml:
'PYTHON' is undefined
##[error]Bash exited with code '1'.
```

File: recipe/meta.yaml [hunk 1]
```diff
@@ -12,15 +12,18 @@ source:
   sha256: {{ sha256 }}
 
 build:
-  number: 0
+  number: 1
   noarch: python
-  script: python setup.py install --single-version-externally-managed --record record.txt
+  script: "{{ PYTHON }} -m pip install . --no-deps -vv"
```
Comment (orbeckst) [inline recipe/meta.yaml:15]: Well, maybe one needs to define `PYTHON` somewhere....

Comment (orbeckst) [reply recipe/meta.yaml:15]: I followed the recipe at https://conda-forge.org/docs/maintainer/adding_pkgs.html#use-pip , thinking that ``PYTHON`` would be automagically defined.

**Comment by orbeckst** [comment]
[These instructions](https://conda-forge.org/docs/maintainer/adding_pkgs.html#use-pip) are not fool-proof... 
```
/home/conda/feedstock_root/build_artifacts/griddataformats_1554697898490/_build_env/bin/python: No module named pip
Traceback (most recent call last):
  File "/opt/conda/bin/conda-build", line 11, in <module>
    sys.exit(main())
  File "/opt/conda/lib/python3.7/site-packages/conda_build/cli/main_build.py", line 456, in main
    execute(sys.argv[1:])
  File "/opt/conda/lib/python3.7/site-packages/conda_build/cli/main_build.py", line 447, in execute
    verify=args.verify, variants=args.variants)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/api.py", line 208, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 2314, in build_tree
    notest=notest,
  File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 1477, in build
    cwd=src_dir, stats=build_stats)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/utils.py", line 374, in check_call_env
    return _func_defaulting_env_to_os_environ('call', *popenargs, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/conda_build/utils.py", line 354, in _func_defaulting_env_to_os_environ
    raise subprocess.CalledProcessError(proc.returncode, _args)
subprocess.CalledProcessError: Command '['/bin/bash', '-e', '/home/conda/feedstock_root/build_artifacts/griddataformats_1554697898490/work/conda_build.sh']' returned non-zero exit status 1.
```

Even though `pip` was installed (see [pip:             19.0.3-py37_0        conda-forge](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=26939&view=logs&jobId=1bf226d3-0e2f-52d8-fa93-7d9e633347b3&taskId=c09a5a47-eec3-5700-9284-6f1bd7b23501&lineStart=352&lineEnd=353&colStart=1&colEnd=1)), apparently, running `python -m pip install ...` fails.

**Comment by orbeckst** [comment]
Now it fails with 
```
ImportError: No module named pip._internal
```

I'll give up for now.

If anyone who has a better understanding of what conda-forge does wants to jump in, please do.

**Comment by orbeckst** [comment]
@conda-forge/core I tried to implement the suggestion by the linter to use a pip install as written in https://conda-forge.org/docs/maintainer/adding_pkgs.html#use-pip but failed because apparently `pip` is not available in the build even though I added it to the `host` requirements. (Furthermore, `{{ PYTHON }}` was not defined and I had it replaced with a bare `python`.) Insights/help are appreciated, thanks.

**Comment by ocefpaf** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,46 +1,43 @@
...
 
 requirements:
-  build:
+  host:
```
Comment (ocefpaf) [inline recipe/meta.yaml:18]: Python and pip should be in host, not build. See https://conda-forge.org/docs/maintainer/adding_pkgs.html#requirements for more info.

PS: the rest of my changes here are minor cosmetic changes.

Comment (orbeckst) [reply recipe/meta.yaml:18]: Many thanks, much appreciated.

**Pull request merged**